src/driver/driver.hpp

```diff
@@ -57,12 +58,15 @@ class Driver {
 class EvolutionDriver : public Driver {
  public:
   EvolutionDriver(ParameterInput *pin, Mesh *pm) : Driver(pin, pm) {
-    Real start_time = pinput->GetOrAddReal("parthenon/time", "start_time", 0.0);
-    Real tstop = pinput->GetReal("parthenon/time", "tlim");
+    Real start_time = pinput->GetOrAddPrecise("parthenon/time", "start_time", 0.0);
+    Real tstop = pinput->GetOrAddPrecise("parthenon/time", "tlim",
+                                         std::numeric_limits<Real>::max());
```

Comment:
I don't like this default floating around in multiple places. Maybe we could put it in `defs` as a `constexpr Real default default_time` or something? Also it really looks to me like what you want here is a sginal NaN, not just the max Real.

Reply:
I disagree. I think it belongs here.  If a time limit is not specified, the simulation is supposed to run 'forever'.  In terms of double precision, that 'forever' is represented by max().

Reply:
Hmm okay. I buy that.

Is there some reason it shouldn't be `std::numeric_limits<Real>::infinity()` then? Infinity would really be forever, as it's not just the largest double, it always compares as bigger than any double, including `std::numeric_limits<Real>::max()`?

Reply:
I'll have to look up the result of comparisons with infinity. Do you happen to know off-hand?

Reply:
I'm pretty sure it works like you intuitively expect. Infinity > any non-inf, non-NaN > -infinity